command/agent/command.go

```diff
@@ -362,6 +399,42 @@ func (config *Config) verifyUniqueListeners() error {
 	return nil
 }
 
+// discoverEc2Hosts searches the given AWS region, returning a list of instance
+// addresses where EC2TagKey = EC2TagValue
+func (c *Config) discoverEc2Hosts() ([]string, error) {
+	config := c.EC2Discovery
+	awsConfig := &aws.Config{
+		Region:      aws.String(config.Region),
+		Credentials: credentials.NewStaticCredentials(config.AccessKeyID, config.SecretAccessKey, ""),
```

Comment:
I ended up using a ChainProvider to match the order in which Terraform's [`AWS Provider`](https://www.terraform.io/docs/providers/aws/index.html) looks for credentials.